core/helpers/EEH_Line_Item.helper.php

```diff
@@ -1749,7 +1749,10 @@ public static function calculate_final_price_for_ticket_line_item(
             );
         }
         // ok now find this new registration's final price
-        if (isset($final_prices_per_ticket_line_item[ $total_line_item->ID() ][ $ticket_line_item->ID() ])) {
+        if (isset(
+            $final_prices_per_ticket_line_item[ $total_line_item->ID() ],
+            $final_prices_per_ticket_line_item[ $total_line_item->ID() ][ $ticket_line_item->ID() ]
```

Comment:
this shouldn't be necessary as PHP checks each index sequentially, so the previous single line of code should do what you want.

from the PHP docs:

```php
$a = array (
    'test' => 1,
    'hello' => NULL,
    'pie' => array(
        'a' => 'apple'
    )
);

var_dump(isset($a['test']));            // TRUE
var_dump(isset($a['foo']));             // FALSE
var_dump(isset($a['hello']));           // FALSE

// The key 'hello' equals NULL so is considered unset
// If you want to check for NULL key values then try: 
var_dump(array_key_exists('hello', $a)); // TRUE

// Checking deeper array values
var_dump(isset($a['pie']['a']));        // TRUE
var_dump(isset($a['pie']['b']));        // FALSE
var_dump(isset($a['cake']['a']['b']));  // FALSE
```

notice in the last block of code that they did not have to do:

```php
isset($a['pie'], $a['pie']['a']);
```

(EDIT)
Just saw your commit note now (wasn't visible from the code review screen) 

> check both the top-level item and the lower item are set. Because isset has a bad habit of giving the wrong result when you ask for both at the same time

really? how so? I have not experienced this (not saying it is not the case).

I looked through the comments on https://www.php.net/manual/en/function.isset.php
and found the following scenarios that may be the kind of cases you are referring to:

 - https://www.php.net/manual/en/function.isset.php#112046
 - https://www.php.net/manual/en/function.isset.php#47396

the first case deals with objects that are using `__isset()` 
and the second case deals with string values getting converted to arrays and returning unexpected results, which I'm guess is from before PHP 5.4 considering the comment is 14 years old.

Are either of these what you are referring to? Or are you seeing something else?

Reply:
oh, ya I got a warning and incorrectly assumed it was because we were checking both the top-level array key and the lower one simultaneously. But like you said, that's actually not the problem. The error only coincidentally went away when I added this change.

(And ya, I was vaguely recalling php.net/manual/en/function.isset.php#112046, but got it mixed up.)

Looking at it more closely, the warning was

> Warning: Illegal offset type in E:\Software\laragon\www\laragonee\wp-content\plugins\event-espresso-core\core\helpers\EEH_Line_Item.helper.php on line 1753

Which wasn't even that exact line... all sorts of oups.

The latest commit fixes the actual problem. For realz this time.